src/js/game/LevelMVC/LevelView.js

```diff
@@ -2096,28 +2099,69 @@ module.exports = class LevelView {
         this.playAnimationWithOffset(sprite, "idle", 29, 1);
         break;
 
-      case "redstoneSparkle":
+      case "redstoneSparkle": {
         atlas = this.blocks[blockType][0];
         frame = this.blocks[blockType][1];
         xOffset = this.blocks[blockType][2];
         yOffset = this.blocks[blockType][3];
-        sprite = group.create(xOffset + 40 * x, yOffset + group.yOffset + 40 * y, atlas, frame);
-        frameList = Phaser.Animation.generateFrameNames("redstone_sparkle", 0, 24, ".png");
-        for (i = 0; i < 4; ++i) {
-          frameList.push("redstone_sparkle7");
+        let posX = xOffset + 40 * x;
+        let posY = yOffset + group.yOffset + 40 * y;
+        sprite = group.create(posX, posY, atlas, frame);
+        // Establish the three different animations.
+        let anim = [];
+        let whichAnim = Math.floor(Math.random() * 3);
+        // Pick one of the animations to assign.
+        switch (whichAnim) {
+          case 0:
+            anim = Phaser.Animation.generateFrameNames("redstone_sparkle", 0, 7, ".png");
+            break;
+          case 1:
+            anim = Phaser.Animation.generateFrameNames("redstone_sparkle", 8, 15, ".png");
+            break;
+          default:
+            anim = Phaser.Animation.generateFrameNames("redstone_sparkle", 16, 23, ".png");
+            break;
         }
-        frameList = frameList.concat(Phaser.Animation.generateFrameNames("redstone_sparkle", 8, 13, ".png"));
-        for (i = 0; i < 3; ++i) {
-          frameList.push("redstone_sparkle13");
+
+        // Set up a random amount of blank frames to be put on the front and back for delay between loops.
+        let frontDelay = Math.floor(Math.random() * 50) + 5;
+        let backDelay = Math.floor(Math.random() * 30) + 20;
+        let frontBuffer = [];
+        let backBuffer = [];
+        for (i = 0; i < frontDelay; ++i) {
+          frontBuffer.push("redstone_sparkle99.png");
         }
-        frameList = frameList.concat(Phaser.Animation.generateFrameNames("redstone_sparkle", 14, 23, ".png"));
-        for (i = 0; i < 8; ++i) {
-          frameList.push("redstone_sparkle1");
+        for (i = 0; i < backDelay; ++i) {
+          backBuffer.push("redstone_sparkle99.png");
         }
-        sprite.animations.add("idle", frameList, 5, true);
+
+        // Organize the animation with a buffer in front and back
+        frameList = frontBuffer.concat(frameList);
+        frameList = frameList.concat(anim);
+        frameList = frameList.concat(backBuffer);
+
+        let animation = sprite.animations.add("idle", frameList, 5, true);
+
+        // onLoop, we want to randomize which corner of the index the animation manifests in.
+        animation.onLoop.add((sprite) => {
+          let xType = Math.floor(Math.random() * 2);
+          let yType = Math.floor(Math.random() * 2);
+          if (xType) {
+            sprite.position.x = sprite.cameraOffset.x + 20;
```

Comment:
What is the `sprite.cameraOffset` value?  My concern is that this might be tied to the current camera location and move on larger 20x20 levels.  Can we calculate the the corner positions from the `x` and `y` values passed in to `createBlock`?

Reply:
Yeeeeeeah, I have that concern as well. What I'm using it for there is that when you go into that anonymous function as a part of the onLoop call, I was losing passed in arguments on subsequent calls. Basically, position changes, so that's not a reliable value to grab in terms of an offset, cameraOffset however was never changing.

Basically, cameraOffset was always the center of the index and I used it as an anchor point to math out where I want the sprite to be drawn (which is captured in position). On subsequent loops, position will have changed and you'd get the sprite just drifting away unless you can reference that original anchor point somehow?

I have the same concern about moving cameras though... but perhaps it'd be fine, because we'd want the sprite to ""move"" like the rest of the index to a different position on the screen? I think this ought to be tested with a big map.

Reply:
@joshlory confirmed that using the camera offset results in the correct behavior.

my guess is that it works something like:
>position means draw to screen with some math: 40 * x, 60 * y (size of each sprite)
>but that's wrong if the camera isn't focused on the same spot, on a huge map 40 * x index, if x were say = 5, shouldn't draw in the center of the player area if we're way off on some corner of the map, so take the camera's position into account.
>sparkle is now doing the same thing, taking the camera position into account.

Now that we're parenting them, I bet you could say something like getParent() -> using parent index in levelModel somehow to find its sprite -> go to actionGroup to find the appropriate parent sprite -> use THAT sprites position as an offset for the child sprite...

But in doing that I imagine you'd just get the same numbers: the position -> screen coords + camera offset.

I think it's fine to skip all that hubbub and just use the math we know is being used anyway? As I said, I mad a 10x13 map and walked around with the camera panning and the sparkle always plays in the right position in relation to the redstoneWire that's generating it. 

👌